lldb/include/lldb/Expression/DWARFExpression.h

```diff
@@ -40,6 +35,31 @@ class DWARFUnit;
 /// location expression or a location list and interprets it.
 class DWARFExpression {
 public:
+  class Delegate {
+  public:
+    Delegate() = default;
+    virtual ~Delegate() = default;
+
+    virtual uint16_t GetVersion() const = 0;
+    virtual dw_addr_t GetBaseAddress() const = 0;
+    virtual uint8_t GetAddressByteSize() const = 0;
+    virtual llvm::Error GetDIEBitSizeAndSign(uint64_t die_offset,
+                                             uint64_t &bit_size,
+                                             bool &sign) = 0;
```

Comment:
```suggestion
    virtual llvm::Expected<std::pair<uint64_t, bool>> GetDIEBitSizeAndSign(uint64_t relative_die_offset) const = 0;
```

(I'm prioritizing the interface over implementation, so I'm making the method const. You can put the const_cast into the implementation (I think it's only needed for `GetDIE`. I am also emphasizing the relativeness of the die offset as lldb default is to use absolute section offsets)

Reply:
Done.